lib/nsync/initializer.coffee

```diff
@@ -197,12 +197,13 @@ module.exports = helper = (activationState) ->
 
   disposables.forEach (disposable) -> composite.add(disposable)
 
-  atomHelper.getToken().then (token) ->
-    nsync.configure
-      expansionState: activationState.directoryExpansionStates
-      localRoot: _path.join(atom.configDirPath, '.learn-ide')
-      connection:
-        url: "#{WS_SERVER_URL}?token=#{token}&version=#{atomHelper.learnIdeVersion()}"
+  atomHelper.waitForTerminalConnection().then ->
+    atomHelper.getToken().then (token) ->
+      nsync.configure
+        expansionState: activationState.directoryExpansionStates
+        localRoot: _path.join(atom.configDirPath, '.learn-ide')
+        connection:
+          url: "#{WS_SERVER_URL}?token=#{token}&version=#{atomHelper.learnIdeVersion()}"
```

Comment:
you can write this as:

```coffeescript
atomHelper.waitForTerminalConnection()
  .then(atomHelper.getToken)
  .then((token) ->
    nsync.configure
```

if you wanna avoid nesting them through anonymous functions like that

you could even then wrap the `nsync.configure` call:

```coffeescript
atomHelper.waitForTerminalConnection()
  .then(atomHelper.getToken)
  .then(nsync.configureFromToken)
```

that function probably doesn't belong on the `nsync` object though since it's a specific detail on how nsync is used in this context